### DISCUSSION TITLE
Styled component wrapper

### DISCUSSION THREAD

**Discussion by southkeys** [jpudysz/react-native-unistyles#310] [Feature Requests]
Hey thanks for the amazing lib 
may have missed it but would be cool to create simple `styled` wrapper that wraps components + passes `createStyleSheet` props
most other libs have this, saves enormous boilerplate for no-logic themed components

e.g., without styled component:
```js
import { View } from 'react-native';
import { ViewProps } from 'react-native-svg/lib/typescript/fabric/utils';
import { createStyleSheet, useStyles } from 'react-native-unistyles';
type Props = {
  isOnline: boolean;
  diameter?: number;
};
const StatusDot = (props: Props & ViewProps) => {
  const { styles } = useStyles(stylesheet);
  return <View style={[styles.container(props), props.style]} />;
};
const stylesheet = createStyleSheet(_theme => ({
  container: ({ isOnline, diameter }: Props) => ({
    height: diameter || 12,
    width: diameter || 12,
    borderRadius: (diameter && diameter / 2) || 6,
    backgroundColor: isOnline ? '#46c69c' : 'transparent',
  }),
}));
```
with styled component
```js
import styled from '@emotion/native';
export const StatusDot = styled(View)(
  (props: { isOnline: boolean; diameter?: number }) => ({
    height: props.diameter || 12,
    width: props.diameter || 12,
    borderRadius: (props.diameter && props.diameter / 2) || 6,
    backgroundColor: props.isOnline ? '#46c69c' : 'transparent',
  }),
);
```

**Comment by jpudysz** [comment]
That's the idea of Unistyles - I don't propose any opinionated solution, allowing you to build whatever you want on top of it.
You can easily create your own wrapper to suit your needs. I don't see the point in repeating styled syntax or maintaining it.

**Comment by southkeys** [comment]
Unistyles is in and of itself *opinionated*, and you have scope creeped a whole slop of features "your own way" (opinionated) that were already implemented by a gazillion other libraries. e.g. Users could also just create functional stylesheet object, use core react-native StatusBar, or expo-navigation-bar, etc so why repeat it and maintain it?

styled components are a dominant paradigm. if you don't agree with the approach, that's ... an opinion. But you are just making it harder for unistyles to break the ice with 
1) developers accustomed to styled components but new to unistyles who just want to get going asap without having to handroll tooling they find ootb in most frameworks. 
2) large refactors involving styled components (as is our case currently: ~600 styled components)

Way to encourage others to make feature requests: suggest their request is pointless

**Comment by jpudysz** [comment]
> styled components are a dominant paradigm

Unistyles won't be a wrapper for styled syntax. It can be if you do it yourself. Just create your own component like in the snippet above. I don't see the point of requesting it and asking me to maintain it. I won't use it, and I don't want to force anyone to use it.
I'm providing low-level primitives, so anyone can build their own design system.

> developers accustomed to styled components but new to unistyles who just want to get going asap without having to handroll tooling they find ootb in most frameworks.

Again, build this component yourself.. you will find it familiar. Just keep in mind that others don't necessarily need it too.

> Large refactors involving styled components (as is our case currently: ~600 styled components)

Changing design system in the middle of the project is not the best idea. 

> Unistyles is in and of itself opinionated, and you have scope creeped a whole slop of features "your own way" (opinionated) that were already implemented by a gazillion other libraries

Yet, here you are using it. This is open source - please fork it and maintain it. It's easy to say "do something because I need it." I'm spending hundreds of hours to make it work. If you don't like it, then again, you have other options.

**Comment by southkeys** [comment]
- **builds opinionated full-fledged styling _framework_**
- **claims is building "primitives"**
- **refuses to acknowledge any form of merit for simple addition to the framework that would make it more competitive**

I just threw styled component out there as an idea. I am not insisting on you to do anything. This was more for you than for me. Unistyles really is not the hot deal you're making it to be.

Furthermore the attitude of dismissing valuable feature requests (or any request made in good faith for that matter) as pointless just cost you a ton of credibility.

Good luck with your project

**Reply by jpudysz** [reply]
Thanks, and good luck to you

**Comment by southkeys** [comment]
wow that was quick! 

https://github.com/jpudysz/react-native-unistyles/commit/3dc63ef93535451dd9c1a88a2aa90affce95540a

thanks ;)
